src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -833,7 +889,7 @@ private static RequiredHelperFunctions EmitTryMatchAtCurrentPosition(IndentedTex
                     // This is the case for single and multiple characters, though the whole thing is only guaranteed
                     // to have been validated in TryFindNextPossibleStartingPosition when doing case-sensitive comparison.
                     writer.WriteLine($"int start = base.runtextpos;");
-                    writer.WriteLine($"int end = start + {(node.Kind == RegexNodeKind.Multi ? node.Str!.Length : 1)};");
+                    writer.WriteLine($"int end = start {((node.Options & RegexOptions.RightToLeft) == 0 ? "+" : "-")} {(node.Kind == RegexNodeKind.Multi ? node.Str!.Length : 1)};");
```

Comment:
> (node.Options & RegexOptions.RightToLeft) == 0

I see that we are making this check in several of the local funcs, but I suppose it can't really be cached since we really mean checking the options of each node as opposed to the original passed in functions right

Reply:
> it can't really be cached since we really mean checking the options of each node as opposed to the original passed in functions right?

Correct

Reply:
and just for my understanding, is the only case where a node's options (in terms of right-to-left) differ from the original passed-in options when using a lookbehind or a conditional lookbehind in the pattern? Or are there other cases?

Reply:
> is the only case where a node's options (in terms of right-to-left) differ from the original passed-in options when using a lookbehind or a conditional lookbehind in the pattern?

Yes.
https://github.com/dotnet/runtime/pull/66280/files#diff-e00e8e70002d7c3c7d0b37a2f661816ec255ce89c86723667e12143c45f71945R1948-R1950